capstone/src/main/java/com/google/step/clients/ScoreClient.java

```diff
@@ -0,0 +1,160 @@
+package com.google.step.clients;
+
+import com.google.step.data.RestaurantPageViews;
+import com.google.step.data.RestaurantScore;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Collections;
+
+
+/**
+ * Will take care of calculating the score for each restaurant.
+ */
+public class ScoreClient {
+
+    private final float minimum = 10.0f;
+    private final MetricsClient metricsClient = new MetricsClient();
+
+    /**
+     * Only method to be called and calculates the scores of each restaurant.
+     * @return an unmodifiable list of restaurantScore class.
+     */
+    public List<RestaurantScore> calculateScores() {
```

Comment:
General comment:

This is something you can do AFTER the presentation if you have time, so don't feel the need to refactor now.

It would likely be a lot more efficient (and possibly easier to read) if you explicitly extracted things into arrays, did the math, and then recombined, as follows:

(forgive broken pseudocode, it's just to give you an idea of the general pattern)

```
// Unpack
Map<Long, Integer> indexes = new HashMap<>();
double scores[] = new double[allPageViews.size()];
int pageViews[] = new int[allPageViews.size()];
int idx = 0;
for (RestaurantPageViews pageView : allPageViews) {
    indexes.add(pageView.getRestaurantKey(), idx);
    scores[idx] = 0.0; // or whatever you need it to be
    pageViews[idx] = pageView.getCount(); // or whatever it is...
    ++idx;  
}

// Run your scoring algorithm on the arrays; you don't need to keep looking up restaurant keys/ids/names, they don't matter for this part.
(...)

// Pack your results as needed.
for (Map::Entity<Long, Integer> idx : indexes) {
    result.add(<something something> idx.Key(), <something something> scores[idx.Value()]);
}
return result;
```

Reply:
I did a similar method to what you explained and didn't even need to use maps! Fixed!